Supporting Otsu threshold

gulshan-123 opened the pull request:
Added an initial version of otsu threshold. It only supports single channel u8 images.
Solves #478 
Implementation according to the algorithm given in the paper: https://www.ipol.im/pub/art/2016/158/article_lr.pdf

edgarriba commented on crates/kornia-imgproc/src/threshold.rs:
```diff
@@ -374,6 +374,66 @@ where
 // TODO: outsu, triangle
 
+/// Apply an otsu threshold to an image
+pub fn threshold_otsu<A1: ImageAllocator, A2: ImageAllocator>(
```
provide a test comparing with an exsiting library to match compatibility

gulshan-123 commented:
Hi @edgarriba, I have added one test for the otsu thresholding. Moreover, I have added to return the otsu threshold value computed.
On my side, I have tested for all the images present on the [demo of paper you sent](https://ipolcore.ipol.im/demo/clientApp/demo.html?id=158&key=FD4EE592DAF214DFEE6AC32108F47310). All are successful, but included only 1 in the code.

The values were confirmed from python opencv implementation, the snippet for checking value is:
```python3
import cv2
image = cv2.imread('input_1.png', cv2.IMREAD_GRAYSCALE)
thresh_value, otsu_img = cv2.threshold(image, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)
print(f"Otsu Threshold: {thresh_value}")
```

edgarriba commented on crates/kornia-imgproc/src/threshold.rs:
```diff
@@ -374,11 +374,72 @@ where
+pub fn threshold_otsu<A1: ImageAllocator, A2: ImageAllocator>(
+    src: &Image<u8, 1, A1>,
+    dst: &mut Image<u8, 1, A2>,
+    threshold_out: &mut u8,
```
why do we need the threshold output ?

edgarriba commented on crates/kornia-imgproc/src/threshold.rs:
```diff
@@ -374,11 +374,72 @@ where
+        ));
+    }
+    let mut hist: Vec<usize> = vec![0_usize; 256];
+    let _ = compute_histogram(src, &mut hist, 256).unwrap();
```
no unwraps

edgarriba commented on crates/kornia-imgproc/src/threshold.rs:
```diff
@@ -374,11 +374,72 @@ where
+    let n = (src.rows() * src.cols()) as f64;
+
+    let mut sum = 0f64;
+    for i in 0..256 {
```
```suggestion
    for i in 0..HIST_SIZE {
```
no magic numbers, use consts. same for the rest

edgarriba commented on crates/kornia-imgproc/src/threshold.rs:
```diff
@@ -374,11 +374,72 @@ where
+        }
+    }
+    *threshold_out = thresh;
+    parallel::par_iter_rows_val(src, dst, |src_pixel, dst_pixel| {
```
there's a `threshold` set of reusable functions

edgarriba commented on tests/data/barabara.png:
now new images to be added for testing for now, we want to keep low the git tree size
